File: android/app/src/main/kotlin/com/example/pdp_todo_app/native/BatteryPigeonApi.kt
package com.example.pdp_todo_app.native

import android.content.Context
import com.example.pdp_todo_app.pigeon.BatteryHostApi
import com.example.pdp_todo_app.pigeon.FlutterError

class BatteryPigeonApi(
    private val context: Context
) : BatteryHostApi {
    override fun getBatteryLevel(): Long {
        val batteryLevel = BatteryLevelReader.capacityPercent(context)
            ?: throw FlutterError(
                "UNAVAILABLE",
                "Battery level not available.",
                null
            )
        return batteryLevel.toLong()
    }
}

File: ios/Runner/Native/BatteryPigeonApi.swift
import UIKit

final class BatteryPigeonApi: BatteryHostApi {
    func getBatteryLevel() throws -> Int64 {
        guard let batteryLevel = BatteryLevelReader.capacityPercent() else {
            throw PigeonError(
                code: "UNAVAILABLE",
                message: "Battery level not available.",
                details: nil
            )
        }
        return Int64(batteryLevel)
    }
}
